packages/integrations/react/src/index.ts

```diff
@@ -1,5 +1,14 @@
 import type { AstroIntegration } from 'astro';
 import { version as ReactVersion } from 'react-dom';
+import react, {type Options as ViteReactPluginOptions} from '@vitejs/plugin-react';
+
+const FAST_REFRESH_PREAMBLE = `
+import RefreshRuntime from '/@react-refresh'
+RefreshRuntime.injectIntoGlobalHook(window)
+window.$RefreshReg$ = () => {}
+window.$RefreshSig$ = () => (type) => type
+window.__vite_plugin_react_preamble_installed__ = true
+`;
```

Comment:
I think we can also get the preamble code from vite plugin react, like `react.preambleCode` ([source code](https://github.com/vitejs/vite-plugin-react/blob/4bebe5bd7c0267f6b088005293870cf69953b73a/packages/plugin-react/src/index.ts#L284)).

We also need to handle the `base` here so the preamble code is correctly loaded on different base. (Above the linked source code has an example)